tests/src/JIT/Regression/JitBlue/DevDiv_707061/DevDiv_707061.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// The test should repro an assert failure when with JitStressRegs=0x3 we can't find a register slot to construct
+// `PUTARG_SPLIT` that needs 5 regs (4 uses and 1 def) but can use only r3-r5, because r0-r3 are
+// not spillable because they are marked with `isBusyUntilNextKill`.
+
+using System;
+using System.Runtime.CompilerServices;
+
+class DevDiv_707061
+{
+
+    struct PromotedWith4Fields
+    {
+        public int a;
+        public int b;
+        public int c;
+        public int d;
+    }
+
+    [MethodImplAttribute(MethodImplOptions.NoInlining)]
+    static void CallWithManyArgs(int r0, int r1, int r2, PromotedWith4Fields s)
+    {
+        s.a = r0;
+        s.b = r1;
+        s.c = r2;
+        int sum = s.a + s.b + s.c + s.d;
+        Console.WriteLine(sum);
+    }
+
+    [MethodImplAttribute(MethodImplOptions.NoInlining)]
+    static void CallWithSomeArgs(int r0, int r1, int r2)
+    {
+
+        Console.WriteLine(r0 + r1 + r2);
+    }
+
+
+    [MethodImplAttribute(MethodImplOptions.NoInlining)]
+    static void Test(int v)
+    {
+        PromotedWith4Fields str;
+        int a = 1 - v;
+        int b = 10 + v;
+        int c = 2 * v;
+        str.a = a;
+        str.b = b;
+        str.c = c;
+        str.d = 4;
+        // a, b and c will generate `PUTARG_REG` that are marked as `isSpecialPutArg`,
+        // we also need to force `LinearScan::allocateRegisters()` to set 
+        // `physRegRecord->isBusyUntilNextKill = true;` for them, then `allocateBusyReg`
+        // won't be able to spill them and won't have enough space to construct `PUTARG_SPLIT`
+        // for str.
+        // However, it doesn't happen in this test, so it needs something else.
+        CallWithManyArgs(a, b, c, str);
+
```

Comment:
LSRA shows that `PUTARG_REG` nodes for this two calls do not lay on the same interval, so thi check in `allocateRegisters` fails:
```
                    if (srcInterval->isActive &&
                        genRegMask(srcInterval->physReg) == currentRefPosition->registerAssignment &&
                        currentInterval->getNextRefLocation() == physRegRecord->getNextRefLocation())
```
and clear `currentInterval->isSpecialPutArg`.

Reply:
The arguments need to be the same local variable, and there must be no uses of that local after the call. This is the tricky situation where a local goes dead at a call, but it has more than one use at the call, so, e.g. if we move the lclVar into the desired register (at which point, since it's dying at this program point, the old reg location could become free), we still need to keep the lclVar live.